config/routes.rb

```diff
@@ -1,6 +1,7 @@
 Rails.application.routes.draw do
   get '/cart', to: 'cart#show'
   post '/cart', to: 'cart#create'
+  post '/search', to: 'restaurants#search'
```

Comment:
Can this be refactored under resources for restaurants?

Reply:
What do you mean?

Reply:
The search action is not a CRUD action, it has to be added as a "custom" action.
If you have "resources restaurants" then there won't be a route for the restaurants controller with an action of search.

Reply:
I would argue that you can still use the `index` acction and pass in a param with the query string. In your controller index action you can have a condifion like 
```
if params[:search] 
 # here you do whatever you want to do with search
else
  # here you do your normal stuff
end
```
That's one way of doing this and staying in the REST routes structure

Reply:
Didn't think about solving it like that, will try that now :)

Reply:
Added the factored version and everything looks good.